### PR TITLE
Amending after PR 17.10.18

### PR DESCRIPTION
**Pull request by Kalyanova**

### CODE REVIEW

File: app/src/main/java/by/paranoidandroid/cityweather/AndroidApplication.kt [hunk 1]
```diff
@@ -2,74 +2,66 @@ package by.paranoidandroid.cityweather
...
         }
     }
 
-    protected fun buildComponent(): AppComponent {
+    private fun buildComponent(): AppComponent {
         return DaggerAppComponent.builder()
-                .api(Api)
                 .appModule(AppModule(this))
                 .roomModule(RoomModule(this))
+                .retrofitModule(RetrofitModule())
```
Comment (aplachykau) [inline app/src/main/java/by/paranoidandroid/cityweather/AndroidApplication.kt:54]: networkmodule will be suitable

File: app/src/main/java/by/paranoidandroid/cityweather/domain/repository/CityRepository.kt [hunk 2]
```diff
@@ -60,79 +62,63 @@ class CityRepository @Inject constructor(val webService: WebRepository, val dbRe
 
     /**
      * @param ids: identifiers of cities
-     * @param oldValues is here to update LiveData only if result isn't null.
      */
     @Suppress(UNCHECKED_CAST)
```
Comment (aplachykau) [inline app/src/main/java/by/paranoidandroid/cityweather/domain/repository/CityRepository.kt:66]: not need use it as constant

File: app/src/main/java/by/paranoidandroid/cityweather/domain/repository/CityRepository.kt [hunk 3]
```diff
@@ -60,79 +62,63 @@ class CityRepository @Inject constructor(val webService: WebRepository, val dbRe
...
-    fun getForecastsFromNetwork(ids: String, oldValues: ForecastList?): LiveData<ForecastList> {
-        Log.d(TAG, "getForecastsFromNetwork")
-        val data = MutableLiveData<ForecastList>()
-        val connManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+    fun getForecastsFromNetwork(ids: String): MutableLiveData<ForecastList> {
```
Comment (aplachykau) [inline app/src/main/java/by/paranoidandroid/cityweather/domain/repository/CityRepository.kt:67]: too complex. try to reduce hierarchy and create separate methods

File: app/src/main/java/by/paranoidandroid/cityweather/injection/RetrofitModule.kt [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+@Module
+class RetrofitModule {
+
+    @Singleton
+    @Provides
+    fun provideRetrofit(): Retrofit {
+        val logInterceptor = HttpLoggingInterceptor()
```
Comment (aplachykau) [inline app/src/main/java/by/paranoidandroid/cityweather/injection/RetrofitModule.kt:21]: provide it as separate dependencies

File: app/src/main/java/by/paranoidandroid/cityweather/injection/RetrofitModule.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                .setLevel(HttpLoggingInterceptor.Level.BASIC)
+
+        val httpClient = OkHttpClient.Builder()
```
Comment (aplachykau) [inline app/src/main/java/by/paranoidandroid/cityweather/injection/RetrofitModule.kt:24]: provide it as separate dependencies

File: app/src/main/java/by/paranoidandroid/cityweather/network/Service.kt [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import by.paranoidandroid.cityweather.injection.RetrofitModule.Companion.APP_ID
+import by.paranoidandroid.cityweather.network.entity.WebCityList
+import by.paranoidandroid.cityweather.network.entity.WebForecast
+import io.reactivex.Single
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface Service {
+
+    @GET("/data/2.5/weather")
```
Comment (aplachykau) [inline app/src/main/java/by/paranoidandroid/cityweather/network/Service.kt:12]: will be good to add docs and examples of JSON format what you expect to receive

File: app/src/main/java/by/paranoidandroid/cityweather/view/fragment/CitiesFragment.kt [hunk 7]
```diff
@@ -4,72 +4,85 @@ import android.arch.lifecycle.Observer
...
             }
         })
+    }
+
+    override fun onCreateView(inflater: LayoutInflater,
+                              container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.fragment_cities, container, false)
+        recyclerViewCities = view.findViewById(R.id.recycler_view_cities)
+        recyclerViewCities?.layoutManager = LinearLayoutManager(context)
```
Comment (aplachykau) [inline app/src/main/java/by/paranoidandroid/cityweather/view/fragment/CitiesFragment.kt:54]: move to xml

File: app/src/main/java/by/paranoidandroid/cityweather/view/fragment/CitiesFragment.kt [hunk 8]
```diff
@@ -4,72 +4,85 @@ import android.arch.lifecycle.Observer
...
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        val view = inflater.inflate(R.layout.fragment_cities, container, false)
-
-        val cityAdapter = CityAdapter(context)
-        view.recyclerViewCities.adapter = cityAdapter
-        view.recyclerViewCities.layoutManager = LinearLayoutManager(context)
+    lateinit var viewModel: CitiesViewModel
+    private var cityAdapter: CityAdapter? = null
+    private var recyclerViewCities: RecyclerView? = null
+    private var progressBar: ContentLoadingProgressBar? = null
```
Comment (aplachykau) [inline app/src/main/java/by/paranoidandroid/cityweather/view/fragment/CitiesFragment.kt:32]: kotlinknife

File: app/src/main/java/by/paranoidandroid/cityweather/AndroidApplication.kt (hunk 1 above)
Comment (Kalyanova) [reply app/src/main/java/by/paranoidandroid/cityweather/AndroidApplication.kt:54]: ✅ Fixed in commits a9052d0435a4372c02d15e95f4195f5bf02748f0 (_"Refactoring (RetrofitModule was renamed to NetworkModule)"_) and 7ef84fc248a708705edd94039029d274d7f70c50 (_"Fix of previous commit (RetrofitModule was renamed to NetworkModule) and refactoring"_)

File: app/src/main/java/by/paranoidandroid/cityweather/domain/repository/CityRepository.kt (hunk 2 above)
Comment (Kalyanova) [reply app/src/main/java/by/paranoidandroid/cityweather/domain/repository/CityRepository.kt:66]: ✅ Fixed in commit 7ef84fc248a708705edd94039029d274d7f70c50 (_"Fix of previous commit (RetrofitModule was renamed to NetworkModule) and refactoring"_)

File: app/src/main/java/by/paranoidandroid/cityweather/view/fragment/CitiesFragment.kt (hunk 7 above)
Comment (Kalyanova) [reply app/src/main/java/by/paranoidandroid/cityweather/view/fragment/CitiesFragment.kt:54]: ✅ Fixed in commit 7ef84fc248a708705edd94039029d274d7f70c50 (_"Fix of previous commit (RetrofitModule was renamed to NetworkModule) and refactoring"_).

`LayoutManager ` already was defined in xml:

```
<android.support.v7.widget.RecyclerView
        android:id="@+id/recycler_view_cities"
       ...
        app:layoutManager="android.support.v7.widget.LinearLayoutManager"
        ... />
```
so I've just removed this unnecessary line in code:

`recyclerViewCities?.layoutManager = LinearLayoutManager(context)`

**Pull request merged**